Rater.Data/Repositories/RatingRepository.cs

```diff
@@ -0,0 +1,60 @@
+﻿using AutoMapper;
+using Rater.API;
+using Rater.Data.DataContext;
+using Rater.Data.Repositories.Interfaces;
+using Rater.Domain.DataTransferObjects.RatingDto;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Rater.Data.Repositories
+{
+    public class RatingRepository : IRatingRepository
+    {
+
+        private readonly DBBContext _context;
+        private readonly IMapper _mapper;
+        public RatingRepository(DBBContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<List<RatingForMetricResponseDto>> AddRatings(List<Rating> request)
+        {
+
+            foreach (var x in request)
+            {
+                var metric = await _context.Metrics.Where(e => e.MetricId == x.MetricId).FirstOrDefaultAsync();
+                var participant = await _context.Participants.Where(e => e.ParticipantId == x.RateeId).FirstOrDefaultAsync();
+
+                if(metric?.SpaceId != x.SpaceId || participant?.SpaceId != x.SpaceId)
+                {
+                    throw new Exception("istekte uyuşmazlık tespit edildi");                                // If participant or metrics are not belong to the given space id.
```

Comment:
Let's continue in English

Reply:
And remove this comment, use method documentations for it. [Check this out](https://chatgpt.com/share/2d176b38-3e90-4d43-8cb2-7ebbf3c71a57) - F1